code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm

```diff
@@ -282,9 +282,19 @@
 
 	stab_direction = turn(get_dir(xeno, targeted_atom), 180)
 	playsound(hit_target,'sound/weapons/alien_tail_attack.ogg', 50, TRUE)
-	if(hit_target.mob_size < MOB_SIZE_BIG)
-		step_away(hit_target, xeno)
 
+	var/direction = Get_Compass_Dir(xeno, targeted_atom) //More precise than get_dir.
+	if(!direction) //Same tile.
+		return
```

Comment:
Why would we not perform damage if we're on the same tile? The only thing that should be skipped is the wallslam part of the code. We have yet to apply the slow, animation, damage, log, etc. yet.